src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl

```diff
@@ -85,8 +85,373 @@
+                if (bi + out_b < BATCH_SIZE)                                \
+                    OUTPUT_BLOCK_WRITE(output, output_offset, result[bi]);  \
+                output_offset += TILE_OUT_B_PITCH;                          \
+            } while (false)
+#else
+        #define WRITE_OUTPUT(bi) do {                                       \
+                OUTPUT_BLOCK_WRITE(output, output_offset, result[bi]);      \
+                output_offset += TILE_OUT_B_PITCH;                          \
+            } while (false)
+#endif
+        CONST_LOOP(FORCED_TILE_B, WRITE_OUTPUT);
+        #undef WRITE_OUTPUT
+    } else {
+        output_offset += sglid;
+
+        // TODO: Investigate why below code doesn't compile and check how it affects performance.
+        //#define WRITE_OUTPUT_FEATURE(fi) do {                                                   \
+        //        const bool should_write =                                                       \
+        //            TILE_OUT_F_NUM %  (TILE_OFM * SIMD) == 0 ||                                 \
+        //            out_f + (fi) * SIMD + sglid < TILE_OUT_F_NUM;                               \
+        //        if (should_write) {                                                             \
+        //            output[output_offset] = result[out_bi][fi];                                 \
+        //        }                                                                               \
+        //        output_offset += SIMD;                                                          \
+        //    } while (false)
+        //
+        //#define WRITE_OUTPUT(bi) do {                                                           \
+        //        const uint out_bi = bi;                                                         \
+        //        CONST_LOOP(TILE_OFM, WRITE_OUTPUT_FEATURE);                                     \
+        //        output_offset += TILE_OUT_B_PITCH - TILE_OFM * SIMD;                            \
+        //    } while (false)
+        //
+        //CONST_LOOP(FORCED_TILE_B, WRITE_OUTPUT);
+        //#undef WRITE_OUTPUT
+        //#undef WRITE_OUTPUT_FEATURE
+
+        for (uint bi = 0; bi < FORCED_TILE_B; ++bi) {
+            for (uint fi = 0; fi < TILE_OFM; ++fi) {
+                const bool should_write =
+#if IS_DYNAMIC
+                    bi + out_b < BATCH_SIZE &&
+#endif
+                    (TILE_OUT_F_NUM % (TILE_OFM * SIMD) == 0 ||
+                    out_f + fi * SIMD + sglid < TILE_OUT_F_NUM);
+                if (should_write) {
+                    output[output_offset] = ((OUTPUT_TYPE*)(&result[bi]))[fi];
+                }
+                output_offset += SIMD;
+            }
+            output_offset += TILE_OUT_B_PITCH - TILE_OFM * SIMD;
+        }
+    }
+    // =====================================================================================================================================
+    #undef FORCED_TILE_B
+}
```

Comment:
And why do we need this separate implementation? Seems that the only difference is FORCED_TILE_B, and I think we can nly set something like ACTUAL_TILE_B  to select either of TILE_B or BATCH_SIZE..?

Reply:
Yeah, the only difference is FORCED_TILE_B. In case of INT4 weights compression the idea is to have two "kernels versions" (actually two independent branches: tuned for high batch number by default by kernel selector (TILE_B=8) and optimized for single batch(TILE_B=1)) at the same time in single kernel
I'm not sure how we can implement it in other way (unless we use one huge macro definition, which I think is not the best idea)
BTW, I think it's a good idea to try same approach for INT8 weights compression as well, because currently TILE_B is forced to 8 there, and probably we can reduce actual work per work item